src/Symfony/Component/Dotenv/Tests/DotenvTest.php

```diff
@@ -228,6 +228,16 @@ public function testHttpVarIsPartiallyOverriden()
         $this->assertSame('http_value', $_SERVER['HTTP_TEST_ENV_VAR']);
     }
 
+    public function testEnvVarIsOverriden()
+    {
+        putenv('TEST_ENV_VAR_OVERRIDEN=original_value');
+
+        $dotenv = new DotEnv(true);
+        $dotenv->populate(array('TEST_ENV_VAR_OVERRIDEN' => 'new_value'));
+
+        $this->assertSame('new_value', getenv('TEST_ENV_VAR_OVERRIDEN'));
```

Comment:
Add some asserts for the `_ENV`, `_SERVER` vars too ?

Reply:
Asserts added.

Reply:
Don't you want to test they're getting properly overridden from `$_SERVER` and `$_ENV`?

Reply:
@fmata